File: Data Structures & Algorithms/kth-largest-integer-in-a-stream/submission-0.java
class KthLargest {
    int k = 0;
    int[] nums;
    public KthLargest(int k, int[] nums) {
        this.k = k;
        this.nums = nums;
        Arrays.sort(nums);
    }
    
    public int add(int val) {
        int[] copy = new int[nums.length+1];
        for(int i = 0; i < nums.length; i++){
            copy[i] = nums[i];
        }
        copy[copy.length-1] = val;

        nums = copy;

        Arrays.sort(nums);
        for(int i : nums){
            System.out.println("Nums: " + i);
        }
        int res = k;
        for(int i = nums.length - 1; i >= 0; i--){
            res--;
            if(res == 0){
                return nums[i];
            }
        }
        return -1;
    }
}

File: Data Structures & Algorithms/permutations-ii/submission-0.py
class Solution:
    def permuteUnique(self, nums: List[int]) -> List[List[int]]:
        res = []
        stack = []
        freqs = {}
        for num in nums:
            freqs[num] = freqs.get(num, 0) + 1
        nums.sort()
        def dfs(i):
            nonlocal res, stack, freqs
            if i == len(nums):
                res.append(stack[:])
                return
            for num in freqs:
                if freqs[num] <= 0:
                    continue
                freqs[num] -= 1
                stack.append(num)
                dfs(i + 1)
                freqs[num] += 1
                stack.pop()
            

        dfs(0)
        return res